src/esa_apex_toolbox/upscaling/udp_job_manager.py

```diff
@@ -0,0 +1,146 @@
+import ast
+from pathlib import Path
+from typing import Optional
+
+import pandas as pd
+import requests
+import shapely
+
+import openeo
+from openeo.extra.job_management import MultiBackendJobManager
+
+
+class UDPJobManager(MultiBackendJobManager):
+    """
+    Large area processing for UDP's.
+
+    This job manager can run complex workflows without requiring project specific dependencies.
+    """
+    
+    def __init__(self, udp_id:str, udp_namespace:str, fixed_parameters:dict, job_options:dict=None):
+        super().__init__()
+        self.largescale_process = None
+        self._job_options = job_options
+        self.fixed_parameters = fixed_parameters
+        self.udp_namespace = udp_namespace
+        self.udp_id = udp_id
+        self.dataframe: pd.DataFrame = None
+
+        self._parse_udp()
+
+    def _parse_udp(self):
+        self.udp_metadata = requests.get(self.udp_namespace).json()
+
+    @property
+    def job_options(self):
+        return self._job_options
+
+    @job_options.setter
+    def job_options(self, value):
+        self._job_options = value
+
+    def udp_parameters(self) -> list[dict]:
+        return self.udp_metadata["parameters"]
+
+    def udp_parameter_schema(self, name:str) -> Optional[dict]:
+        return {p["name"]:p.get("schema",None) for p in self.udp_parameters()}.get(name,None)
+
+
+    def add_jobs(self, jobs_dataframe):
+        """
+        Add jobs to the job manager.
+
+        Column names of the dataframe have to match with UDP parameters.
+
+        Extra columns names:
+
+        - `title` : Title of the job
+        - `description` : Description of the job
+
+        """
+        if self.dataframe is None:
+            self.dataframe = jobs_dataframe
+        else:
+            raise ValueError("Jobs already added to the job manager.")
```

Comment:
this `if else raise` pattern looks like this could have been a constructor argument